### PR TITLE
add alerting v2beta1 apis

### PR DESCRIPTION
**Pull request by junotx**
### What type of PR is this?

/kind feature

### What this PR does / why we need it:

alerting v2beta1 apis will operate the resources of `RuleGroup`, `ClusterRuleGroup`, `GlobalRuleGroup` objects: 
- common operations may call the k8s apis for these resources.
- getting status and alerts of the rules may call the ks apis which this pr contains.

### Which issue(s) this PR fixes:

Fixes #

### Special notes for reviewers:
```
```

### Does this PR introduced a user-facing change?

```release-note
Add alerting v2beta1 apis based on rulegroups resources
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/api/alerting/v2beta1/types.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+type GlobalRuleGroup struct {
+	alertingv2beta1.GlobalRuleGroup `json:",inline"`
+	Status                          RuleGroupStatus `json:"status,omitempty"`
+}
+
+type RuleGroupStatus struct {
+	State          string       `json:"state,omitempty" description:"state of a rule based on its alerts, one of firing, pending, inactive"`
+	EvaluationTime *float64     `json:"evaluationTime,omitempty" description:"taken seconds for evaluation"`
+	LastEvaluation *time.Time   `json:"lastEvaluation,omitempty" description:"time for last evaluation"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:67]:
```suggestion
	LastEvaluation *time.Time   `json:"lastEvaluation,omitempty" description:"time of last evaluation"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+	RuleStatuses   []RuleStatus `json:"ruleStatuses,omitempty" description:"rule statuses"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:68]:
```suggestion
	RulesStatus   []RuleStatus `json:"rulesStatus,omitempty" description:"status of rules in one RuleGroup"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	Status                           RuleGroupStatus `json:"status,omitempty"`
+}
+
+type GlobalRuleGroup struct {
+	alertingv2beta1.GlobalRuleGroup `json:",inline"`
+	Status                          RuleGroupStatus `json:"status,omitempty"`
+}
+
+type RuleGroupStatus struct {
+	State          string       `json:"state,omitempty" description:"state of a rule based on its alerts, one of firing, pending, inactive"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:65]:
```suggestion
	State          string       `json:"state,omitempty" description:"state of a rule, one of firing, pending or inactive depending on its alerts"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	EvaluationTime *float64     `json:"evaluationTime,omitempty" description:"taken seconds for evaluation"`
+	LastEvaluation *time.Time   `json:"lastEvaluation,omitempty" description:"time for last evaluation"`
+	RuleStatuses   []RuleStatus `json:"ruleStatuses,omitempty" description:"rule statuses"`
+}
+
+type RuleStatus struct {
+	State                     string     `json:"state,omitempty" description:"state of a rule based on its alerts, one of firing, pending, inactive"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:72]:
```suggestion
	State                     string     `json:"state,omitempty" description:"state of a rule, one of firing, pending or inactive depending on its alerts"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	Health                    string     `json:"health,omitempty" description:"health state of a rule based on the last execution, one of ok, err, unknown"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:73]:
```suggestion
	Health                    string     `json:"health,omitempty" description:"health state of a rule, one of ok, err or unknown depending on the last evaluation result"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+	LastError                 string     `json:"lastError,omitempty" description:"error for the last execution"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:74]:
```suggestion
	LastError                 string     `json:"lastError,omitempty" description:"error for the last evaluation"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+	EvaluationDurationSeconds float64    `json:"evaluationTime,omitempty" description:"taken seconds for evaluation of query expression"`
+	LastEvaluation            *time.Time `json:"lastEvaluation,omitempty" description:"time for last evaluation of query expression"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:76]:
```suggestion
	LastEvaluation            *time.Time `json:"lastEvaluation,omitempty" description:"time of the last evaluation of the query expression"`
```

File: pkg/api/alerting/v2beta1/types.go [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+
+	Alerts []*Alert `json:"alerts,omitempty" description:"alerts"`
+}
+
+type Alert struct {
+	ActiveAt    *time.Time        `json:"activeAt,omitempty" description:"time when alert is active"`
```
Comment (benjaminhuo) [inline pkg/api/alerting/v2beta1/types.go:82]:
```suggestion
	ActiveAt    *time.Time        `json:"activeAt,omitempty" description:"time when this alert became active"`
```

File: pkg/kapis/alerting/v2beta1/register.go [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+
+	ws.Route(ws.GET("/namespaces/{namespace}/rulegroups").
+		To(handler.handleListRuleGroups).
+		Doc("list the rulegroups in the specified namespace").
+		Param(ws.QueryParameter(query.ParameterName, "name used to do filtering").Required(false)).
+		Param(ws.QueryParameter(query.ParameterPage, "page").Required(false).DataFormat("page=%d").DefaultValue("page=1")).
+		Param(ws.QueryParameter(query.ParameterLimit, "limit").Required(false)).
+		Param(ws.QueryParameter(query.ParameterAscending, "sort parameters, e.g. reverse=true").Required(false).DefaultValue("ascending=false")).
+		Param(ws.QueryParameter(query.ParameterOrderBy, "sort parameters, e.g. orderBy=createTime")).
+		Param(ws.QueryParameter(kapialertingv2beta1.FieldState, "state of a rulegroup based on its rules, one of `firing`, `pending`, `inactive`")).
```
Comment (benjaminhuo) [inline pkg/kapis/alerting/v2beta1/register.go:50]:
```suggestion
		Param(ws.QueryParameter(kapialertingv2beta1.FieldState, "state of a rulegroup, one of `firing`, `pending`, `inactive` depending on its rules")).
```

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

File: pkg/api/alerting/v2beta1/types.go [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+type GlobalRuleGroup struct {
+	alertingv2beta1.GlobalRuleGroup `json:",inline"`
+	Status                          RuleGroupStatus `json:"status,omitempty"`
+}
+
+type RuleGroupStatus struct {
+	State          string       `json:"state,omitempty" description:"state of a rule based on its alerts, one of firing, pending, inactive"`
+	EvaluationTime *float64     `json:"evaluationTime,omitempty" description:"taken seconds for evaluation"`
```
[the remark this answers is not in the document]
Comment (junotx) [reply pkg/api/alerting/v2beta1/types.go]: The `EvaluationTime`, refering to [prometheus api](https://github.com/prometheus/prometheus/blob/v2.36.0/web/api/v1/api.go#L1183), may not change

**Pull request merged**